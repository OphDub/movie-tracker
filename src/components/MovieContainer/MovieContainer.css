.MovieContainer {
  display: flex;
  flex-direction: row;
  background: rgba(100,100,255, 0.1);
}