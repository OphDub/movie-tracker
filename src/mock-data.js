 export const cleanMovieArray = [
    {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    },
    {
      "movie_id": 284054,
      "title": "Black Panther",
      "vote_average": 5,
      "vote_count": 72,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg"
    }
] 

export const dirtyMovieObject = { results: [
    { 
      vote_average: 7.4,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      vote_count: 794,
      title: 'The Shape of Water',
      poster_path: '/iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg',
      release_date: '2017-12-08',
      id: 399055,
      randomnumber: 24341324142134,
    },
    {
      "title": "Black Panther",
      "vote_count": 72,
      "thing": "asdfasdfasdf",
      "id": 284054,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "/bLBUCtMQGJclH36clliPLmljMys.jpg",
      "vote_average": 5,
    }
]}

export const oneMovie = {
  movie_id: 284054,
  title: 'Coco',
  vote_average: 7.7,
  vote_count: 1987,
  overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel\'s family history.',
  posterImage: 'https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg'
}

export const userData = {
  id: 2,
  name: 'brophus park',
  email: 'bros@gmail.com',
  password: 'pass',
}

export const movieRawData = {
  data: {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    }
}
