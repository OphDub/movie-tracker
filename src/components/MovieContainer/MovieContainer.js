import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addFavorite, getAllFavorites } from '../../api';
import './MovieContainer.css';
import Card from '../Card/Card';

export class MovieContainer extends Component {
  cardsArray = () => {
    const moviesArray = this.props.movieArray.map((movie, index) => 
      <Card movie={movie}
            id={Date.now() + index}
            key={index}
            handleFavorite={this.handleFavorite}
       />
    )
    return moviesArray
  }

  handleFavorite = async (movie) => {
    
    if (this.props.activeUser) {
      const user_id = this.props.activeUser.id
      movie.user_id = user_id
      const allFavs = await getAllFavorites(user_id);

      const match = allFavs.data.filter(favMovie => favMovie.movie_id === movie.movie_id)
      console.log("duplicate" , match);
      if(match.length > 0) {
        console.log('remove from faves');
      } else {
        addFavorite(movie) 
      }

    } else {
      this.props.history.push('login')
    }
  }

  render() {
    return (
      <div className="MovieContainer">
        {this.cardsArray()}
      </div>
    )
  }
}

const mapStateToProps = ({movieArray, activeUser}) => ({
    movieArray,
    activeUser
  })

export default connect(mapStateToProps)(MovieContainer)