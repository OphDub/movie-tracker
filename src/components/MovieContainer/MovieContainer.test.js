/* eslint-disable */
import React from 'react';
import { MovieContainer, mapStateToProps, mapDispatchToProps } from './MovieContainer';
import { shallow } from 'enzyme';
import { cleanMovieArray, userData, oneMovie, movieRawData } from '../../mock-data.js';

describe('MovieContainer', () => {
  let wrapper;
  const mockFunction = jest.fn();
  
  beforeEach( () => {
    wrapper = shallow(
      <MovieContainer
        movieArray={cleanMovieArray} 
        favoriteArray={cleanMovieArray}
        activeUser={userData}
        getFavorites={mockFunction}
        handleFavorite={mockFunction}
        addFavorite={mockFunction}
      />);
  });

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should make a fetch call when getFavorites is called', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(movieRawData)
    }))

    wrapper.instance().getFavorites(userData.id)
    wrapper.update()

    expect(window.fetch).toHaveBeenCalled()
  })

  it('should call toggleFavorite when handleFavorite is called', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve(movieRawData)
    }))

    await wrapper.instance().handleFavorite(oneMovie)
    wrapper.update()

    expect(mockFunction).toHaveBeenCalled()
  })

  describe('mapStateToProps and mapDispatchToProps for MovieContainer', () => {
    it('should map the store correctly', () => {
      const mockUser = { id: 4, name: 'bruce', password: 'pass' };
      const mockStore = {
        activeUser: mockUser
      };

      const mapped = mapStateToProps(mockStore);

      expect(mapped.activeUser).toEqual(mockStore.activeUser);
    });

    it('should call the dispatch func when sendFavorite is called', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);

      mapped.addFavorite();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});