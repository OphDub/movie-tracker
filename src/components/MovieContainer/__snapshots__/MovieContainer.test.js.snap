// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieContainer
    activeUser={
      Object {
        "email": "bros@gmail.com",
        "id": 2,
        "name": "brophus park",
        "password": "pass",
      }
    }
    addFavorite={[MockFunction]}
    favoriteArray={
      Array [
        Object {
          "movie_id": 399055,
          "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
          "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
          "release_date": "2017-12-08",
          "title": "The Shape of Water",
          "vote_average": 7.4,
          "vote_count": 794,
        },
        Object {
          "movie_id": 284054,
          "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
          "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
          "release_date": "2018-02-13",
          "title": "Black Panther",
          "vote_average": 5,
          "vote_count": 72,
        },
      ]
    }
    getFavorites={[MockFunction]}
    handleFavorite={[MockFunction]}
    movieArray={
      Array [
        Object {
          "movie_id": 399055,
          "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
          "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
          "release_date": "2017-12-08",
          "title": "The Shape of Water",
          "vote_average": 7.4,
          "vote_count": 794,
        },
        Object {
          "movie_id": 284054,
          "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
          "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
          "release_date": "2018-02-13",
          "title": "Black Panther",
          "vote_average": 5,
          "vote_count": 72,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addClass="favorited"
          handleFavorite={[Function]}
          id={399055}
          movie={
            Object {
              "movie_id": 399055,
              "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
              "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
              "release_date": "2017-12-08",
              "title": "The Shape of Water",
              "vote_average": 7.4,
              "vote_count": 794,
            }
          }
        />,
        <Card
          addClass="favorited"
          handleFavorite={[Function]}
          id={284054}
          movie={
            Object {
              "movie_id": 284054,
              "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
              "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": 5,
              "vote_count": 72,
            }
          }
        />,
      ],
      "className": "MovieContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "399055",
        "nodeType": "function",
        "props": Object {
          "addClass": "favorited",
          "handleFavorite": [Function],
          "id": 399055,
          "movie": Object {
            "movie_id": 399055,
            "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
            "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
            "release_date": "2017-12-08",
            "title": "The Shape of Water",
            "vote_average": 7.4,
            "vote_count": 794,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "284054",
        "nodeType": "function",
        "props": Object {
          "addClass": "favorited",
          "handleFavorite": [Function],
          "id": 284054,
          "movie": Object {
            "movie_id": 284054,
            "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
            "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
            "release_date": "2018-02-13",
            "title": "Black Panther",
            "vote_average": 5,
            "vote_count": 72,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addClass="favorited"
            handleFavorite={[Function]}
            id={399055}
            movie={
              Object {
                "movie_id": 399055,
                "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
                "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
                "release_date": "2017-12-08",
                "title": "The Shape of Water",
                "vote_average": 7.4,
                "vote_count": 794,
              }
            }
          />,
          <Card
            addClass="favorited"
            handleFavorite={[Function]}
            id={284054}
            movie={
              Object {
                "movie_id": 284054,
                "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
                "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
                "release_date": "2018-02-13",
                "title": "Black Panther",
                "vote_average": 5,
                "vote_count": 72,
              }
            }
          />,
        ],
        "className": "MovieContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "399055",
          "nodeType": "function",
          "props": Object {
            "addClass": "favorited",
            "handleFavorite": [Function],
            "id": 399055,
            "movie": Object {
              "movie_id": 399055,
              "overview": "An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.",
              "poster_path": "https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg",
              "release_date": "2017-12-08",
              "title": "The Shape of Water",
              "vote_average": 7.4,
              "vote_count": 794,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "284054",
          "nodeType": "function",
          "props": Object {
            "addClass": "favorited",
            "handleFavorite": [Function],
            "id": 284054,
            "movie": Object {
              "movie_id": 284054,
              "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
              "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": 5,
              "vote_count": 72,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
